### ISSUE TITLE
A very weird issue between this mod and CustomNPCs.

### ISSUE THREAD

**Issue by Koopliss** [DimensionalDevelopment/VanillaFix#130]
So this bug is kinda... strange.

In the CustomNPCs mod i can make an NPC, then i use the cloning tool to clone it. after cloning it and placing it somewhere else, it spawns, then disappears from existence. after leaving the area and coming back, they return.

But this bug gets weirder, it seems that the NPC will forever be removed from existence almost instantly when i spawn a clone i made in the overworld in the nether. no matter what i did they never return in terms of nether clones.

after removing vanillafix, everything was working perfectly. i figured i'd report the incompatibility here to perhaps find some sort of solution.

**Comment by Sunconure11** [comment]
I encountered something similar, but I did not think VanillaFix was to blame.

@Noppes

**Comment by Sunconure11** [comment]
Yeah, it's basically the same issue. I did find a workaround. Use the mob spawner.

**Comment by Koopliss** [comment]
thanks, this bug has been a bit of a major pain and the few workarounds that were found/suggested are also a little bit more... inconvenient than the cloning tool.

**Comment by Sunconure11** [comment]
Is there anything that can be done on this end to solve it?

**Comment by Sunconure11** [comment]
Issue decided to resurface on my end.

**Comment by mcenderdragon** [comment]
Please provide game logs, it sounds like a entity-UUID problem.

**Comment by mcenderdragon** [comment]
If I am right due to the cloning the exact same NPC (same UUID and stuff) exists in the world at the same time, this is not allowed by vanilla and it prints it to the log. The results are server side only entities that show up as sson as on entity dies so only entity with of said UUID is visible in the world at once.

**Comment by Sunconure11** [comment]
I do have some broken NPCs I can use for this, however.

**Comment by Sunconure11** [comment]
Having a hard time finding the log that might contain this, even after some testing with clearly broken NPCs.

**Comment by wideblaze** [comment]
I have this custom npc issue as well -- had no idea it might be related to vanilla fix
